File: frontend/src/types/audienceSegment.ts
export type SegmentType = 'AGE' | 'GENDER' | 'REGION' | 'INTEREST' | 'BEHAVIOR' | 'CUSTOM'

export interface AudienceSegment {
  id: number
  name: string
  type: SegmentType
  criteria: SegmentCriteria
  size: number
  percentage: number
  avgWatchTime: number
  avgRetention: number
  avgCtr: number
  avgEngagement: number
  growthRate: number
  revenueContribution: number
  topContentCategories: string[]
  bestPostingTimes: { day: string; hour: number; score: number }[]
  createdAt: string
  updatedAt: string
}

export interface SegmentCriteria {
  ageRange?: { min: number; max: number }
  gender?: 'MALE' | 'FEMALE' | 'ALL'
  regions?: string[]
  interests?: string[]
  platforms?: string[]
  minWatchTime?: number
  minEngagement?: number
}

export interface SegmentInsight {
  segmentId: number
  contentRecommendations: { category: string; reason: string; confidence: number }[]
  titleStyle: string
  thumbnailStyle: string
  optimalLength: { min: number; max: number; unit: string }
  toneRecommendation: string
  growthOpportunities: string[]
}

export interface SegmentComparison {
  segments: AudienceSegment[]
  metrics: { metric: string; values: { segmentId: number; value: number }[] }[]
}

export interface CreateSegmentRequest {
  name: string
  type: SegmentType
  criteria: SegmentCriteria
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/audiencesegment/AudienceSegment.kt
package com.ongo.domain.audiencesegment

import java.time.LocalDateTime

data class AudienceSegment(
    val id: Long? = null,
    val userId: Long,
    val name: String,
    val type: String = "CUSTOM",
    val criteria: String = "{}",
    val size: Long = 0,
    val percentage: Double = 0.0,
    val avgWatchTime: Double = 0.0,
    val avgRetention: Double = 0.0,
    val avgCtr: Double = 0.0,
    val avgEngagement: Double = 0.0,
    val growthRate: Double = 0.0,
    val revenueContribution: Double = 0.0,
    val topContentCategories: String = "[]",
    val bestPostingTimes: String = "[]",
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null,
)
